lib/typing/values/string_test.go

```diff
@@ -16,274 +15,118 @@ import (
-			},
-			expectedString: `[1,2,3,4,5]`,
-		},
-		{
-			name: "array w/ nested objects (JSON)",
-			colKind: columns.Column{
-				KindDetails: typing.Array,
-			},
-			colVal: []map[string]any{
-				{
-					"dusty": "the mini aussie",
-				},
-				{
-					"robin": "tang",
-				},
-				{
-					"foo": "bar",
-				},
-			},
-			expectedString: `[{"dusty":"the mini aussie"},{"robin":"tang"},{"foo":"bar"}]`,
-		},
-		{
-			name: "array w/ bools",
-			colKind: columns.Column{
-				KindDetails: typing.Array,
-			},
-			colVal: []bool{
-				true,
-				true,
-				false,
-				false,
-				true,
-			},
-			expectedString: `[true,true,false,false,true]`,
-		},
+	{
+		// Struct
+		val, err := ToString(map[string]any{"foo": "bar"}, columns.Column{KindDetails: typing.Struct}, nil)
+		assert.NoError(t, err)
+		assert.Equal(t, `{"foo":"bar"}`, val)
+
+		val, err = ToString(constants.ToastUnavailableValuePlaceholder, columns.Column{KindDetails: typing.Struct}, nil)
+		assert.NoError(t, err)
+		assert.Equal(t, `{"key":"__debezium_unavailable_value"}`, val)
 	}
-
-	for _, testCase := range testCases {
-		evaluateTestCase(t, testCase)
+	{
+		// Array
+		val, err := ToString([]string{"foo", "bar"}, columns.Column{KindDetails: typing.Array}, nil)
+		assert.NoError(t, err)
+		assert.Equal(t, `["foo","bar"]`, val)
 	}
-}
-
-// TestCastColValStaging_Time - will test all the variants of date, time and date time.
-func TestCastColValStaging_Time(t *testing.T) {
-	birthday := time.Date(2022, time.September, 6, 3, 19, 24, 942000000, time.UTC)
-	// date
-	dateKind := typing.ETime
-	dateKind.ExtendedTimeDetails = &ext.Date
-	// time
-	timeKind := typing.ETime
-	timeKind.ExtendedTimeDetails = &ext.Time
-	// date time
-	dateTimeKind := typing.ETime
-	dateTimeKind.ExtendedTimeDetails = &ext.DateTime
-
-	birthdate, err := ext.NewExtendedTime(birthday, dateKind.ExtendedTimeDetails.Type, "")
-	assert.NoError(t, err)
-
-	birthTime, err := ext.NewExtendedTime(birthday, timeKind.ExtendedTimeDetails.Type, "")
-	assert.NoError(t, err)
-
-	birthDateTime, err := ext.NewExtendedTime(birthday, dateTimeKind.ExtendedTimeDetails.Type, "")
-	assert.NoError(t, err)
-
-	testCases := []_testCase{
-		{
-			name:   "date",
-			colVal: birthdate,
-			colKind: columns.Column{
-				KindDetails: dateKind,
-			},
-			expectedString: "2022-09-06",
-		},
-		{
-			name:   "date (but value is dateTime)",
-			colVal: birthDateTime,
-			colKind: columns.Column{
-				KindDetails: dateKind,
-			},
-			expectedString: "2022-09-06",
-		},
-		{
-			name:   "time",
-			colVal: birthTime,
-			colKind: columns.Column{
-				KindDetails: timeKind,
-			},
-			expectedString: "03:19:24.942",
-		},
-		{
-			name:   "datetime",
-			colVal: birthDateTime,
-			colKind: columns.Column{
-				KindDetails: dateTimeKind,
-			},
-			expectedString: "2022-09-06T03:19:24.942Z",
-		},
+	{
+		// Integer
+		// Floats first.
+		val, err := ToString(float32(45452.999991), columns.Column{KindDetails: typing.Integer}, nil)
```

Comment:
Should we be erroring in this case rather than supporting float -> int since it'll result in data loss?

Reply:
There is precision loss as we're rounding up. A hard failure here doesn't necessarily make sense either as most customers would just create a float compatible column in the destination.

Reply:
Another reason for this to happen in a legitimate cause could be that they are inconsistent about passing in the optional schema and JSON marshal made these values into `floats` but they're actually ints.